File: cs/f01.base/Student.cs
public class Student : Person
{
    private double gpa;

    // Constructors
    public Student(string name) : base(name)              { /* nothing */ }             
    public Student(string name, double gpa) : base(name)  { this.GPA = gpa; }

    // Properties
    public double GPA {
        get { return gpa; }    // accessor - get data
        set { gpa = value; }   // mutator  - set data
    }
}

File: javascripts/js/g10.parent/Student.js
load('Person.js');
/******* CLASS DEFINITION *******/
var Student = (function () {
  Student.prototype = new Person();

  /******* CONSTRUCTOR *******/
  function Student (name, gpa) {
    this.name = name;                    // parent is not being accessed.
    this.gpa = gpa != null ? gpa : 0.0;
  }

  /******* PROPERTIES *******/
  Object.defineProperty(Student.prototype, "gpa", {
    get: function(){ return this._gpa },
    set: function(value) { this._gpa = value }
  });

  return Student;
})(Person);
